File: problems/tier2-intermediate/012-elevator-system/boilerplate/javascript/part2/guided.js
// Data class (given).
class ElevOp {
  constructor(kind, s1 = "", i1 = 0, i2 = 0) {
    this.kind = kind;
    this.s1 = s1;
    this.i1 = i1;
    this.i2 = i2;
  }
}

// HINT: introduce an abstraction so new rules don't change existing code.

// HINT: start from what this must return, then work backwards to the state it needs.
function elevator_simulate(ops) {
  // TODO: write your solution
  return null;
}

// ── Export everything the test runner needs (do not remove) ──
// If you add classes (e.g. strategy subclasses), add them here too.
module.exports = { ElevOp, elevator_simulate };

File: problems/tier2-intermediate/014-splitwise/boilerplate/java/part1/Guided.java
import java.util.*;

// Data class (given).
class SplitOp {
    public String kind;
    public String s1;
    public String s2;
    public String s3;
    public String s4;
    public int i1;

    public SplitOp(String kind, String s1, String s2, String s3, String s4, int i1) {
        this.kind = kind;
        this.s1 = s1;
        this.s2 = s2;
        this.s3 = s3;
        this.s4 = s4;
        this.i1 = i1;
    }

    public SplitOp(String kind) {
        this(kind, "", "", "", "", 0);
    }
}

// HINT: introduce an abstraction so new variants don't change existing code.
public class Solution {
    // HINT: start from what this must return, then work backwards to the state it needs.
    public static List<String> splitwise_simulate(List<SplitOp> ops) {
        // TODO: write your solution
        return null;
    }

}
